cfpb/capital-framework
# Add gitbook docs

Sets up the docs in [GitBook](https://gitbook.com/), which removes ruby dependency of jekyll and adds search and the ability to link to docs for editing direct from the docs.

Adds /docs/ directory, which can [host the docs](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/) on GH pages if this is merged into the master branch as well.

Docs are symlinked to the usage docs so they are used directly. They also pull in the current CF version into the docs header as set in package.json.

## Additions

- /docs/ directory to host GitBooks project.
- `gulp docs:build` and `gulp docs:start` commands.

## Changes

- Adjusts docs to fit with GitBooks markdown parsing.
- Uses prism for syntax highlighting.

## Testing

1. Run `npm install`
1. Run `gulp docs:build`
2. Run `gulp docs:start`
3. Visit the docs at http://localhost:4000
4. Edit usage.md docs in /src/cf-buttons or whatever to see that they live reload.
4. You can try changing the version set in `package.json` and re-running step 1-3 to see that it auto-pulls in the version number into the docs heading.

## Screenshots

![2017-12-28-21-23-localhost_4000](https://user-images.githubusercontent.com/704760/34428108-5d3f99a0-ec17-11e7-928f-425b47c1756c.png)

Search example:
![screen shot 2017-12-28 at 9 24 24 pm](https://user-images.githubusercontent.com/704760/34428111-68418bf6-ec17-11e7-997f-9a468da52882.png)

## Notes

- `SUMMARY.md` is auto-generated by `auto-summary` plugin.
- Uses freshly created https://github.com/cfpb/gitbook-plugin-theme-cfpb, which we may be able to customize further to use the old doc layout. But I wanted to get an MVP in that eliminated the gh-pages clone requirement for docs.
